esphome/components/debug/debug_esp32.cpp

```diff
@@ -30,109 +30,162 @@ static const char *const TAG = "debug";
 
 std::string DebugComponent::get_reset_reason_() {
   std::string reset_reason;
-  switch (rtc_get_reset_reason(0)) {
-    case POWERON_RESET:
-      reset_reason = "Power On Reset";
+  switch (esp_reset_reason()) {
+    case ESP_RST_POWERON:
+      reset_reason = "Reset due to power-on event";
       break;
+    case ESP_RST_EXT:
+      reset_reason = "Reset by external pin";
+      break;
+    case ESP_RST_SW:
+      reset_reason = "Software reset via esp_restart";
+      break;
+    case ESP_RST_PANIC:
+      reset_reason = "Software reset due to exception/panic";
+      break;
+    case ESP_RST_INT_WDT:
+      reset_reason = "Reset (software or hardware) due to interrupt watchdog";
+      break;
+    case ESP_RST_TASK_WDT:
+      reset_reason = "Reset due to task watchdog";
+      break;
+    case ESP_RST_WDT:
+      reset_reason = "Reset due to other watchdogs";
+      break;
+    case ESP_RST_DEEPSLEEP:
+      reset_reason = "Reset after exiting deep sleep mode";
+      break;
+    case ESP_RST_BROWNOUT:
+      reset_reason = "Brownout reset (software or hardware)";
+      break;
+    case ESP_RST_SDIO:
+      reset_reason = "Reset over SDIO";
+      break;
+#ifdef USE_ESP32_VARIANT_ESP32
```

Comment:
```suggestion
#ifdef USE_ESP32
```
Required to pass the build checks